cpp/src/arrow/visit_array_inline.h

```diff
@@ -27,11 +27,14 @@ namespace arrow {
   case TYPE_CLASS##Type::type_id:                                                        \
     return visitor->Visit(                                                               \
         internal::checked_cast<const typename TypeTraits<TYPE_CLASS##Type>::ArrayType&>( \
-            array));
+            array),                                                                      \
+        args...);
```

Comment:
Wouldn't it be better to use something like `std::forward<decltype(args)>`?

Reply:
Definitely! Changed to std::forward